### PR TITLE
Performance fixes

### PR DESCRIPTION
**Pull request by lfrancke**
# Description

- HTTP_CLIENT is now static and includes a connection timeout
- StackableAuthorizer only constructs a single instance of StackableAccessControlEnforcer
- StackableAccessControlEnforcer does not serialize to JSON twice anymore
- Spotless formatting applied to code

## Definition of Done Checklist

```[tasklist]
# Reviewer
- [ ] Changelog updated
```

```[tasklist]
# Acceptance
- [ ] Proper release label has been added
```

### CODE REVIEW

File: src/main/java/tech/stackable/hadoop/StackableAccessControlEnforcer.java [hunk 1]
```diff
@@ -19,153 +25,202 @@
...
-        LOG.debug("Started HdfsOpaAccessControlEnforcer");
+public class StackableAccessControlEnforcer
+    implements INodeAttributeProvider.AccessControlEnforcer {
+
+  private static final Logger LOG = LoggerFactory.getLogger(StackableAccessControlEnforcer.class);
+
+  public static final String OPA_POLICY_URL_PROP = "hadoop.security.authorization.opa.policy.url";
+
+  private static final HttpClient HTTP_CLIENT =
+      HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(30)).build();
```
Comment (sbernauer) [inline src/main/java/tech/stackable/hadoop/StackableAccessControlEnforcer.java:57]: Was there any need for a custom timeout or is this "just" considered best practice?

File: src/main/java/tech/stackable/hadoop/StackableAuthorizer.java [hunk 2]
```diff
@@ -7,26 +7,30 @@
...
+  @Override
+  public INodeAttributes getAttributes(String[] strings, INodeAttributes iNodeAttributes) {
+    // No special attributes needed
+    return iNodeAttributes;
+  }
+
+  @Override
+  public AccessControlEnforcer getExternalAccessControlEnforcer(
+      AccessControlEnforcer defaultEnforcer) {
+    return ENFORCER;
```
Comment (sbernauer) [inline src/main/java/tech/stackable/hadoop/StackableAuthorizer.java:34]: Ahhhhh that makes too much sense :upside_down_face:

**Review by sbernauer — APPROVED** [review]
Noice work!

File: src/main/java/tech/stackable/hadoop/StackableAccessControlEnforcer.java (hunk 1 above)
Comment (lfrancke) [reply src/main/java/tech/stackable/hadoop/StackableAccessControlEnforcer.java:57]: "just" best practice

**Pull request merged**